app/services/TwoStepVerification.scala

```diff
@@ -39,37 +38,43 @@ trait TwoStepVerification {
 
   def twoStepVerificationThrottle: TwoStepVerificationThrottle
 
-  case class Biz2SVRule(name: String, conditions: List[Condition])
+  def twoStepVerificationRuleFactory = new TwoStepVerificationUserSegmentFactory {}
 
-  private val biz2svRule = Biz2SVRule("sa", List(not(HasStrongCredentials), GGEnrolmentsAvailable, HasOnlyOneEnrolment, HasSelfAssessmentEnrolments, not(HasRegisteredFor2SV)))
+  private case class Biz2SVRule(name: String, conditions: List[Condition], adminLocations: ThrottleLocations, assistantLocations: ThrottleLocations)
 
-  val continueToAccountUrl = s"${ExternalUrls.taxAccountRouterHost}/account"
+  private val commonRules = List(not(HasRegisteredFor2SV), not(HasStrongCredentials), GGEnrolmentsAvailable)
+
+  private val biz2svRules = twoStepVerificationRuleFactory.rules.map { rule =>
+    Biz2SVRule(rule.name, commonRules ++ List(HasOnlyEnrolments(rule.enrolmentCategories)), rule.adminLocations, rule.assistantLocations)
+  }
 
   def getDestinationVia2SV(continue: Location, ruleContext: RuleContext, auditContext: TAuditContext)(implicit authContext: AuthContext, request: Request[AnyContent], hc: HeaderCarrier) = {
 
+    def throttleLocation(rule: Biz2SVRule) = ruleContext.isAdmin.map(if (_) rule.adminLocations else rule.assistantLocations)
+
     if (twoStepVerificationEnabled && continue == BusinessTaxAccount) {
-      val shouldRedirectTo2SV = biz2svRule.conditions.foldLeft(Future.successful(true)) { (preconditionsTrue, condition) =>
-        preconditionsTrue.flatMap { precondition =>
-          if (!precondition) Future.successful(false)
-          else condition.evaluate(authContext, ruleContext, auditContext)
-        }
-      }
-      shouldRedirectTo2SV.map {
-        case true =>
-          twoStepVerificationThrottle.registrationMandatory(authContext.user.oid) match {
+
+      val applicableRule = biz2svRules.findOne(_.conditions.forAll(_.evaluate(authContext, ruleContext, auditContext)))
+
+      applicableRule.flatMap {
+        case Some(biz2svRule) =>
+          twoStepVerificationThrottle.registrationMandatory(biz2svRule.name, authContext.user.oid) match {
             case true =>
-              auditContext.setSentToMandatory2SVRegister(biz2svRule.name)
-              Some(wrapLocationWith2SV(continue, Locations.TaxAccountRouterHome))
+              throttleLocation(biz2svRule).map { location =>
```

Comment:
This `map` can wrap the `match` above, so that is can be removed from here and the case below